src/clj/witan/send/validate_model.clj

```diff
@@ -0,0 +1,126 @@
+(ns witan.send.validate-model
+  (:require [clojure.data.csv :as csv]
+            [clojure.java.io :as io]
+            [clojure.string :as str]
+            [witan.send.acceptance.workspace-test :refer [run-model]]))
+
+
+(defn temp-dir [input-path]
+  (str "data/" input-path "temp/"))
+
+(defn csv-data->maps [csv-data]
+  (map zipmap
+       (->> (first csv-data)
+            (map keyword)
+            repeat)
+       (rest csv-data)))
+
+(defn write-csv [file data]
+  (with-open [writer (io/writer (io/file file))]
+    (let [columns (into [] (keys (first data)))
+          headers (mapv name columns)
+          rows (mapv #(mapv % columns) data)]
+    (csv/write-csv writer (into [headers] rows)))))
+
+(defn load-csv-as-maps [file]
+  (csv-data->maps
+    (with-open [reader (io/reader file)]
+      (doall
+        (csv/read-csv reader)))))
+
+(defn get-validation-years [transitions]
+  (->> (map :calendar-year transitions)
+       (distinct)
+       (map #(Integer/parseInt %))
+       (drop-last)))
+
+(defn return-fold [bound-fn year transitions]
+  (filter #(bound-fn (Integer/parseInt (:calendar-year %)) year) transitions))
+
+(defn copy-if-exists [source-path destination-path file]
+  (let [source (str source-path file)]
+    (if (.exists (clojure.java.io/as-file source))
+      (io/copy (io/file source) (io/file (str destination-path file))))))
+
+(defn copy-input-files [input-path year]
+  (let [source-path (str "data/" input-path)
+        destination-path (str (temp-dir input-path) year "/")]
+    (.mkdir (java.io.File. destination-path))
+    (doseq [file ["modify-settings.csv" "population.csv" "need-setting-costs.csv" "valid-setting-academic-years.csv"]]
+      (copy-if-exists source-path destination-path file))))
+
+(defn move-target-files [input-path year]
+  (doseq [file ["Output_Count.csv" "Output_AY_State.csv"]]
+    (io/copy (io/file (str "target/" file)) (io/file (str (temp-dir input-path) year "_" file)))))
+
+(defn return-testable-data [file test-years]
+  (->> (for [year test-years] (filter #(= year (:calendar-year %)) (load-csv-as-maps file)))
+       (mapcat seq)))
+
+(defn append-count-with-test [model-count test-data n-transitions]
+  (let [test-year (str (dec (Integer/parseInt (:calendar-year model-count))))
+        test-data-for-year  (filter #(= test-year (:calendar-year %)) test-data)
+        send-test-data-for-year  (filter #(not= "NONSEND" (:need-2 %)) test-data-for-year)
+        ground-truth-count (count send-test-data-for-year)]
+    (assoc model-count :ground-truth (str ground-truth-count) :n-transition n-transitions)))
+
+(defn append-state-with-test [model-state test-data n-transitions]
+  (let [test-year (str (dec (Integer/parseInt (:calendar-year model-state))))
+        state-vector (str/split (str/replace (:state model-state) ":" "") #"-")
```

Comment:
I think you could probably assign it to need and setting and one go using destructing (like in python)
[x,y] (str/split ...)

Reply:
changed